test/ssl_test_ctx.c

```diff
@@ -480,6 +480,39 @@ __owur static int parse_expected_client_cert_type(SSL_TEST_CTX *test_ctx,
                                    value);
 }
 
+/*************************/
+/* Expected signing hash */
+/*************************/
+
+__owur static int parse_expected_sign_hash(int *ptype, const char *value)
+{
+    int nid;
+
+    if (value == NULL)
+        return 0;
+    nid = OBJ_sn2nid(value);
+    if (nid == NID_undef)
+        nid = OBJ_ln2nid(value);
+    if (nid == NID_undef)
```

Comment:
With &&, it means that nid will be assigned the result from `NID_ln2nid()` even if `NID_sn2nid()` returned a valid nid, and if `NID_ln2nid()` happens to return `NID_undef`, this function will return 0 (i.e. fail) even though `NID_sn2nid()` already returned a valid nid.